crates/ere-dockerized/src/lib.rs

```diff
@@ -0,0 +1,512 @@
+        let workspace_dir = workspace_dir();
+
+        let force_rebuild = env::var("ERE_FORCE_REBUILD_DOCKER_IMAGE")
+            .ok()
+            .and_then(|s| s.parse::<bool>().ok())
+            .unwrap_or_default();
+
+        if force_rebuild || !docker_image_exists(self.base_tag(CRATE_VERSION))? {
+            DockerBuildCmd::new()
+                .file(
+                    workspace_dir
+                        .join("docker")
+                        .join("base")
+                        .join("Dockerfile.base"),
+                )
+                .tag(self.base_tag(CRATE_VERSION))
+                .tag(self.base_tag("latest"))
+                .exec(&workspace_dir)
+                .map_err(CommonError::DockerBuildCmd)?;
+        }
+
+        if force_rebuild || !docker_image_exists(self.base_zkvm_tag(CRATE_VERSION))? {
+            DockerBuildCmd::new()
+                .file(
+                    workspace_dir
+                        .join("docker")
+                        .join(self.as_str())
+                        .join("Dockerfile"),
+                )
+                .tag(self.base_zkvm_tag(CRATE_VERSION))
+                .tag(self.base_zkvm_tag("latest"))
+                .bulid_arg("BASE_IMAGE_TAG", self.base_tag(CRATE_VERSION))
+                .exec(&workspace_dir)
+                .map_err(CommonError::DockerBuildCmd)?;
+        }
+
+        if force_rebuild || !docker_image_exists(self.cli_zkvm_tag(CRATE_VERSION))? {
+            DockerBuildCmd::new()
+                .file(workspace_dir.join("docker").join("cli").join("Dockerfile"))
+                .tag(self.cli_zkvm_tag(CRATE_VERSION))
+                .tag(self.cli_zkvm_tag("latest"))
+                .bulid_arg("BASE_ZKVM_IMAGE_TAG", self.base_zkvm_tag(CRATE_VERSION))
+                .bulid_arg("ZKVM", self.as_str())
+                .exec(&workspace_dir)
+                .map_err(CommonError::DockerBuildCmd)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl FromStr for ErezkVM {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "jolt" => Self::Jolt,
+            "nexus" => Self::Nexus,
+            "openvm" => Self::OpenVM,
+            "pico" => Self::Pico,
+            "risc0" => Self::Risc0,
+            "sp1" => Self::SP1,
+            "zisk" => Self::Zisk,
+            _ => return Err(format!("Unsupported zkvm {s}")),
+        })
+    }
+}
+
+impl Display for ErezkVM {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.as_str())
+    }
+}
+
+pub struct EreDockerizedCompiler(pub ErezkVM);
+
+/// Wrapper for serialized program.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct SerializedProgram(Vec<u8>);
+
+impl Compiler for EreDockerizedCompiler {
+    type Error = CompileError;
+    type Program = SerializedProgram;
+
+    fn compile(&self, guest_directory: &Path) -> Result<Self::Program, Self::Error> {
```

Comment:
How would the workload repo use this now?

For example, `guest_directory`would be [this path](https://github.com/eth-act/zkevm-benchmark-workload/tree/master/ere-guests/stateless-validator/sp1)? How it decides to mount the [underlying workspace](https://github.com/eth-act/zkevm-benchmark-workload/tree/master/ere-guests)?

Reply:
Ah I think we could keep it similar with the original one, but make it an instance method like `fn compile(&self, workspace_directory: &Path, guest_relative: &Path)`, so `EreDockerizedCompiler(ErezkVM)` could work.

Reply:
Makes sense, so then `EreDockerizedCompiler` won't impl `Compiler`?

Reply:
BTW, feel free to merge this PR as is -- we can continue the convo in other PR.

Reply:
So in the end I did this https://github.com/eth-act/ere/pull/75/commits/58575d91d6e1d8dc1deb216e537339180761976e, I think this should be flexible enough to support any kinds of layout (actually similar as previous, but now we have 2 absolute path `mounting_directory` and `guest_directory` and require the latter is sub-dir of the former).